File: backend/identity-service/src/main/java/com/skillinfinity/identity/service/AuthService.java
package com.skillinfinity.identity.service;

import com.skillinfinity.identity.dto.request.ChangePasswordRequest;
import com.skillinfinity.identity.dto.request.LoginRequest;
import com.skillinfinity.identity.dto.request.RefreshTokenRequest;
import com.skillinfinity.identity.dto.request.RegisterRequest;
import com.skillinfinity.identity.dto.response.AuthResponse;
import com.skillinfinity.identity.dto.response.TokenValidationResponse;
import com.skillinfinity.identity.dto.response.UserInfoResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    void logout(String refreshToken);

    AuthResponse refreshToken(RefreshTokenRequest request);

    TokenValidationResponse validateToken(String token);

    UserInfoResponse getCurrentUser(String userId);

    void changePassword(String userId, ChangePasswordRequest request);

    void forgotPassword(String email);

    void resetPassword(String token, String newPassword);
}

File: frontend/src/services/auth.service.ts
import type { AxiosRequestConfig } from 'axios';
import { apiClient } from '@/api';
import { API_ENDPOINTS } from '@/constants';
import type {
  ApiResponse,
  AuthResponse,
  AuthUser,
  ChangePasswordRequest,
  LoginRequest,
  RegisterRequest,
} from '@/types';

export const authService = {
  login: (payload: LoginRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>(API_ENDPOINTS.AUTH.LOGIN, payload),

  register: (payload: RegisterRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>(API_ENDPOINTS.AUTH.REGISTER, payload),

  logout: (refreshToken: string) =>
    apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.LOGOUT, { refreshToken }),

  getCurrentUser: (config?: AxiosRequestConfig) =>
    apiClient.get<ApiResponse<AuthUser>>(API_ENDPOINTS.AUTH.ME, config),

  changePassword: (payload: ChangePasswordRequest) =>
    apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.CHANGE_PASSWORD, payload),

  forgotPassword: (email: string) =>
    apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.FORGOT_PASSWORD, null, {
      params: { email },
    }),

  resetPassword: (token: string, newPassword: string) =>
    apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.RESET_PASSWORD, null, {
      params: { token, newPassword },
    }),
};
